pkg/compactor/shard_aware_deduplicate_filter.go

```diff
@@ -31,9 +31,10 @@ func NewShardAwareDeduplicateFilter() *ShardAwareDeduplicateFilter {
 	return &ShardAwareDeduplicateFilter{}
 }
 
-// Filter filters out duplicate blocks that can be formed
-// from two or more overlapping blocks that fully submatches the source blocks of the older blocks.
+// Filter filters out from metas, the initial map of blocks, all the blocks that are contained in other, compacted, blocks from metas.
```

Comment:
"from metas" occurs twice in the sentence, let's dedup :)
```suggestion
// Filter filters out from metas, the initial map of blocks, all the blocks that are contained in other, compacted, blocks.
```